jdaviz/configs/default/plugins/markers/tests/test_markers_plugin.py

```diff
@@ -0,0 +1,240 @@
+    # y = [28, 92] Jy
+    label_mouseover._viewer_mouse_event(sv,
+                                        {'event': 'mousemove',
+                                         'domain': {'x': 4.623e-7, 'y': 0}})
+
+    assert label_mouseover.as_text() == ('Cursor 4.62300e-07, 0.00000e+00 Value +8.00000e+00 Jy',
+                                         'Wave 4.62280e-07 m (0 pix)',
+                                         'Flux 2.80000e+01 Jy')
+    assert label_mouseover.as_dict() == {'data_label': 'test[FLUX]',
+                                         'x': 4.622800069238093e-07,
+                                         'x:unit': 'm',
+                                         'slice index': 0.0,
+                                         'slice wavelength': 4.622800069238093e-07,
+                                         'slice wavelength:unit': 'm',
+                                         'y': 28.0,
+                                         'y:unit': 'Jy'}
+
+    mp._obj._on_viewer_key_event(sv, {'event': 'keydown',
+                                      'key': 'm'})
+    assert len(mp.export_table()) == 2
+    assert len(_get_markers_from_viewer(fv).x) == 1
+    assert len(_get_markers_from_viewer(sv).x) == 1
+
+    # test event in spectrum viewer (with cursor only)
+    label_mouseover.dataset_selected = 'none'
+    label_mouseover._viewer_mouse_event(sv,
+                                        {'event': 'mousemove',
+                                         'domain': {'x': 4.623e-7, 'y': 0}})
+
+    assert label_mouseover.as_text() == ('Cursor 4.62300e-07, 0.00000e+00 Value +8.00000e+00 Jy',
+                                         '', '')
+    assert label_mouseover.as_dict() == {'x': 4.623e-07,
+                                         'x:unit': 'm',
+                                         'y': 0,
+                                         'y:unit': 'Jy',
+                                         'data_label': ''}
+
+    mp._obj._on_viewer_key_event(sv, {'event': 'keydown',
+                                      'key': 'm'})
+    assert len(mp.export_table()) == 3
+    assert len(_get_markers_from_viewer(fv).x) == 1
+    assert len(_get_markers_from_viewer(sv).x) == 2
+
+    # markers hide when plugin closed
+    cubeviz_helper.app.state.drawer = False
+    assert _get_markers_from_viewer(fv).visible is False
+    assert _get_markers_from_viewer(sv).visible is False
+
+    # markers re-appear when plugin re-opened
+    mp.open_in_tray()
+    assert _get_markers_from_viewer(fv).visible is True
+    assert _get_markers_from_viewer(sv).visible is True
+    assert len(_get_markers_from_viewer(fv).x) == 1
+    assert len(_get_markers_from_viewer(sv).x) == 2
+
+    # clearing table clears markers
+    mp.clear_table()
+    assert mp.export_table() is None
+    assert len(_get_markers_from_viewer(fv).x) == 0
+    assert len(_get_markers_from_viewer(sv).x) == 0
+
+
+class TestImvizMultiLayer(BaseImviz_WCS_NoWCS):
```

Comment:
Maybe you should use this test case instead. It loads 3 images that would also let you test WCS linking.

https://github.com/spacetelescope/jdaviz/blob/d2fde6be6cbe564c6cf4a4d0abede2b23e87014c/jdaviz/configs/imviz/tests/utils.py#L98

https://github.com/spacetelescope/jdaviz/blob/d2fde6be6cbe564c6cf4a4d0abede2b23e87014c/jdaviz/configs/imviz/tests/utils.py#L138-L144